lib/explorer/series.ex

```diff
@@ -251,6 +251,7 @@ defmodule Explorer.Series do
     * `:dtype` - Cast the series to a given `:dtype`. By default this is `nil`, which means
       that Explorer will infer the type from the values in the list.
       See the module docs for the list of valid dtypes and aliases.
+    * `:strict` - when `true` raises on overflow and underflow - defaults to `false`.
```

Comment:
When it is false, what happens? The value is simply discarded?

Reply:
I am a bit confused because it seems Rustler will raise on overflow-underflow if `strict: false`.

Reply:
When false overflow happens. Discard only happens when a series is created as {:s, 64} and cast it to {:s, 8}. And cast does not happen with existing code with and without strict.
```
Series.from_list([-129, 256], dtype: {:s, 16}) |> Series.cast({:s, 8})
#Explorer.Series<
  Polars[2]
  s8 [nil, nil]
>
```
More details about rustler behaviour - https://github.com/elixir-explorer/explorer/pull/826#issuecomment-1902647024